app/models/annotations/status.rb

```diff
@@ -105,6 +105,10 @@ def destroy
 
   private
 
+  def set_annotator
+    self.annotator = User.current unless User.current.nil?
+  end
+
```

Comment:
I agree with Sawy: this is actually a behaviour of the "singleton" annotation (of which status is the only example currently). Eventually, we will move singleton-specific code to a child class of Annotation that Status will inherit from.